class2/processes_DIY.livemd

```diff
@@ -83,5 +198,55 @@ graph LR;
 ```
 
 ```elixir
+defmodule TheRing do
+  def chinese_whispers(name, next_whisperer) do
+    receive do
+      :exit ->
+        send(next_whisperer, :exit)
+        IO.puts(name <> " exited gracefully")
+        :ok
+
+      recived_whisper ->
+        IO.puts(name <> " received: " <> recived_whisper)
+        Process.sleep(1000)
+        send(next_whisperer, recived_whisper)
+        chinese_whispers(name, next_whisperer)
+    end
+  end
+
+  def setup(name) do
+    receive do
+      next_whisperer ->
+        chinese_whispers(name, next_whisperer)
+    end
+  end
+
+  def spawn_nodes(0, last_pid) do
+    last_pid
+  end
+
+  def spawn_nodes(number, last_pid) do
+    new_pid = spawn(TheRing, :chinese_whispers, ["#{number}", last_pid])
+    spawn_nodes(number - 1, new_pid)
+  end
```

Comment:
Again, this is a great way to do it, I just wanted to mention that an alternative approach would be to have each process spawning it's a follower in the line.

Reply:
up